internal/services/mysql/mysql_flexible_server_resource.go

```diff
@@ -506,6 +506,7 @@ func resourceMysqlFlexibleServerRead(d *pluginsdk.ResourceData, meta interface{}
 			d.Set("zone", props.AvailabilityZone)
 			d.Set("version", string(pointer.From(props.Version)))
 			d.Set("fqdn", props.FullyQualifiedDomainName)
+			d.Set("create_mode", d.Get("create_mode").(string))
```

Comment:
In other SQL related resources we set this field during Import, since this is an Import specific issue - can we do that here too?

Reply:
I remember previously you suggested to read the value from tf config when API doesn't return the value. May I ask if  we need to follow your above suggestion for this kind of issue?

Reply:
@tombuildsstuff , I remember previously setting "d.Set("create_mode", d.Get("create_mode").(string))" in read func can always set the property with the value in the tf config to handle the issue of not returning the value by service API but seems it doesn't work anymore. I tried below fixes but all they don't work. Is there any suggestion about how to fix the issue of API doesn't return the value?

Fix # 1. I tried to add "d.Set("create_mode", d.Get("create_mode").(string))" in read func but TF still always sets it as "null" in state file when API doesn't return the value.

Fix # 2. I tried to add d.set() with d.Get() to importer but TF still always sets it as "null" in state file when API doesn't return the value.
```
func mysqlFlexibleServerResourceImporter(ctx context.Context, d *pluginsdk.ResourceData, meta interface{}) ([]*pluginsdk.ResourceData, error) {
	d.Set("create_mode", d.Get("create_mode"))

	return []*pluginsdk.ResourceData{d}, nil
}
```

Reply:
For this kind of issue, seems only ignore_changes can workaround this issue for now. So I will close this PR.

Reply:
@neil-yechenwei as I mentioned in my original comment, if you refer to the other places where we set `create_mode` at import time we set a value for this field: https://github.com/hashicorp/terraform-provider-azurerm/blob/2e7a6136cdd0a500950cf536a1684ca3214bfc80/internal/services/mssql/mssql_database_resource.go#L129-L132 (and [override it in some cases](https://github.com/hashicorp/terraform-provider-azurerm/blob/2e7a6136cdd0a500950cf536a1684ca3214bfc80/internal/services/mssql/mssql_database_resource.go#L123) depending on the import type, although it appears the ordering is a bug here).

As such this is possible to provide a default value for this field, and then override it from the config as needed.

Reply:
> @neil-yechenwei as I mentioned in my original comment, if you refer to the other places where we set `create_mode` at import time we set a value for this field:
> 
> https://github.com/hashicorp/terraform-provider-azurerm/blob/2e7a6136cdd0a500950cf536a1684ca3214bfc80/internal/services/mssql/mssql_database_resource.go#L129-L132
> 
> (and [override it in some cases](https://github.com/hashicorp/terraform-provider-azurerm/blob/2e7a6136cdd0a500950cf536a1684ca3214bfc80/internal/services/mssql/mssql_database_resource.go#L123) depending on the import type, although it appears the ordering is a bug here).
> As such this is possible to provide a default value for this field, and then override it from the config as needed.

Here "the config" you mentioned points to the tf config file or the configuration returned by service API.

Reply:
@neil-yechenwei

> I assume 

Rather than assuming, please validate the API behaviour for each type of CreateMode.

Whilst you're correct insofar as `PointInTimeRestore` and `GeoRestore` are going to be similar, the API responses mentioned above should allow us to do _something_ similar to this:

```
			// CreateMode defaults to `Default` but can be overridden, as such we need to check the
			// API Response to determine which to set.
			createMode := servers.CreateModeDefault
			if model := resp.Model; model != nil {
				if props := model.Properties; props != nil {
					if props.RestorePointInTime != nil {
						createMode = servers.CreateModePointInTimeRestore
					}

					if props.ReplicationRole != nil && *props.ReplicationRole == servers.ReplicationRoleReplica {
						createMode = servers.CreateModeReplica
					}
				}
			}
			d.Set("create_mode", string(createMode))
```

As such would you be able to log the API responses in each case here, so that we can determine how to proceed?

Thanks!

Reply:
@tombuildsstuff , Below is the test result. Except "create_mode = Replica", seems we can't distinguish other create modes from the service configuration.

Below is the response when creating mysql fs with create_mode = "Default" or not set create_mode.
```
{
  "sku": {
    "name": "Standard_D4ds_v4",
    "tier": "GeneralPurpose"
  },
  "properties": {
    "administratorLogin": "adminTerraform",
    "storage": {
      "storageSizeGB": 20,
      "iops": 360,
      "autoGrow": "Enabled",
      "autoIoScaling": "Disabled",
      "storageSku": "Premium_LRS",
      "logOnDisk": "Disabled"
    },
    "version": "8.0.21",
    "state": "Ready",
    "fullyQualifiedDomainName": "acctest-fs-test20.mysql.database.azure.com",
    "availabilityZone": "2",
    "maintenanceWindow": {
      "customWindow": "Disabled",
      "dayOfWeek": 0,
      "startHour": 0,
      "startMinute": 0
    },
    "replicationRole": "None",
    "replicaCapacity": 10,
    "network": {
      "publicNetworkAccess": "Enabled"
    },
    "backup": {
      "backupRetentionDays": 7,
      "geoRedundantBackup": "Disabled",
      "earliestRestoreDate": "2024-03-14T02:27:47.0636752+00:00"
    },
    "highAvailability": {
      "mode": "Disabled",
      "state": "NotEnabled",
      "standbyAvailabilityZone": ""
    },
    "privateEndpointConnections": []
  },
  "location": "West Europe",
  "tags": {},
  "id": "/subscriptions/xx-xx-xxx-xx/resourceGroups/acctestRG-mysql-test20/providers/Microsoft.DBforMySQL/flexibleServers/acctest-fs-test20",
  "name": "acctest-fs-test20",
  "type": "Microsoft.DBforMySQL/flexibleServers"
}
```

Below is the response when creating mysql fs with create_mode = "PointInTimeRestore".
```
{
  "sku": {
    "name": "Standard_D4ds_v4",
    "tier": "GeneralPurpose"
  },
  "properties": {
    "administratorLogin": "adminTerraform",
    "storage": {
      "storageSizeGB": 20,
      "iops": 360,
      "autoGrow": "Enabled",
      "autoIoScaling": "Disabled",
      "storageSku": "Premium_LRS",
      "logOnDisk": "Disabled"
    },
    "version": "8.0.21",
    "state": "Ready",
    "fullyQualifiedDomainName": "acctest-fs-pitr-test20.mysql.database.azure.com",
    "availabilityZone": "2",
    "maintenanceWindow": {
      "customWindow": "Disabled",
      "dayOfWeek": 0,
      "startHour": 0,
      "startMinute": 0
    },
    "replicationRole": "None",
    "replicaCapacity": 10,
    "network": {
      "publicNetworkAccess": "Enabled"
    },
    "backup": {
      "backupRetentionDays": 7,
      "geoRedundantBackup": "Disabled",
      "earliestRestoreDate": "2024-03-14T02:48:05.905809+00:00"
    },
    "highAvailability": {
      "mode": "Disabled",
      "state": "NotEnabled",
      "standbyAvailabilityZone": ""
    },
    "privateEndpointConnections": []
  },
  "location": "West Europe",
  "tags": {},
  "id": "/subscriptions/xx-xx-xxx-xx/resourceGroups/acctestRG-mysql-test20/providers/Microsoft.DBforMySQL/flexibleServers/acctest-fs-pitr-test20",
  "name": "acctest-fs-pitr-test20",
  "type": "Microsoft.DBforMySQL/flexibleServers"
}
```

Below is the response when creating mysql fs with create_mode = "Replica".
```
{
  "sku": {
    "name": "Standard_D4ds_v4",
    "tier": "GeneralPurpose"
  },
  "properties": {
    "administratorLogin": "adminTerraform",
    "storage": {
      "storageSizeGB": 20,
      "iops": 360,
      "autoGrow": "Enabled",
      "autoIoScaling": "Disabled",
      "storageSku": "Premium_LRS",
      "logOnDisk": "Disabled"
    },
    "version": "8.0.21",
    "state": "Ready",
    "fullyQualifiedDomainName": "acctest-fs-replica-test20.mysql.database.azure.com",
    "sourceServerResourceId": "/subscriptions/xx-xx-xxx-xx/resourceGroups/acctestRG-mysql-test20/providers/Microsoft.DBforMySQL/flexibleServers/acctest-fs-test20",
    "availabilityZone": "2",
    "maintenanceWindow": {
      "customWindow": "Disabled",
      "dayOfWeek": 0,
      "startHour": 0,
      "startMinute": 0
    },
    "replicationRole": "Replica",
    "replicaCapacity": 0,
    "network": {
      "publicNetworkAccess": "Enabled"
    },
    "backup": {
      "backupRetentionDays": 7,
      "geoRedundantBackup": "Disabled",
      "earliestRestoreDate": "2024-03-14T03:10:34.5234034+00:00"
    },
    "highAvailability": {
      "mode": "Disabled",
      "state": "NotEnabled",
      "standbyAvailabilityZone": ""
    },
    "privateEndpointConnections": []
  },
  "location": "West Europe",
  "tags": {},
  "id": "/subscriptions/xx-xx-xxx-xx/resourceGroups/acctestRG-mysql-test20/providers/Microsoft.DBforMySQL/flexibleServers/acctest-fs-replica-test20",
  "name": "acctest-fs-replica-test20",
  "type": "Microsoft.DBforMySQL/flexibleServers"
}
```

Below is the response when creating mysql fs with create_mode = "GeoRestore".
```
{
  "sku": {
    "name": "Standard_D4ds_v4",
    "tier": "GeneralPurpose"
  },
  "properties": {
    "administratorLogin": "adminTerraform",
    "storage": {
      "storageSizeGB": 20,
      "iops": 360,
      "autoGrow": "Enabled",
      "autoIoScaling": "Disabled",
      "storageSku": "Premium_LRS",
      "logOnDisk": "Disabled"
    },
    "version": "8.0.21",
    "state": "Ready",
    "fullyQualifiedDomainName": "acctest-fs-restore-test20.mysql.database.azure.com",
    "availabilityZone": "2",
    "maintenanceWindow": {
      "customWindow": "Disabled",
      "dayOfWeek": 0,
      "startHour": 0,
      "startMinute": 0
    },
    "replicationRole": "None",
    "replicaCapacity": 10,
    "network": {
      "publicNetworkAccess": "Enabled"
    },
    "backup": {
      "backupRetentionDays": 7,
      "geoRedundantBackup": "Disabled",
      "earliestRestoreDate": "2024-03-14T03:43:34.9680531+00:00"
    },
    "highAvailability": {
      "mode": "Disabled",
      "state": "NotEnabled",
      "standbyAvailabilityZone": ""
    },
    "privateEndpointConnections": []
  },
  "location": "North Europe",
  "tags": {},
  "id": "/subscriptions/xx-xx-xxx-xx/resourceGroups/acctestRG-mysql-test20/providers/Microsoft.DBforMySQL/flexibleServers/acctest-fs-restore-test20",
  "name": "acctest-fs-restore-test20",
  "type": "Microsoft.DBforMySQL/flexibleServers"
}
```